packages/core/src/exchanges/cache.ts

```diff
@@ -131,10 +140,18 @@ export const afterMutation = (
   const pendingOperations = new Set<number>();
   const { additionalTypenames } = response.operation.context;
 
-  [
+  const invalidTypeNames = [
     ...collectTypesFromResponse(response.data),
     ...(additionalTypenames || []),
-  ].forEach(typeName => {
+  ];
+
+  client.eventTarget.dispatchEvent({
```

Comment:
if we can reduce this to `dispatchEvent` we can easily write a `babel-plugin` that wraps this with `__dev__` like we use in [`graphCache`](https://github.com/FormidableLabs/urql/blob/master/scripts/babel/transform-invariant-warning.js)